src/xsdba/adjustment.py

```diff
@@ -0,0 +1,1965 @@
+            Time series to be bias-adjusted, usually a model output.
+        \*\*kwargs
+            Algorithm-specific keyword arguments, see class doc.
+
+        Returns
+        -------
+        xr.Dataset
+            The bias-adjusted Dataset.
+        """
+        if sim is None:
+            sim = hist.copy()
+            sim.attrs["_is_hist"] = True
+
+        kwargs = parse_group(cls._adjust, kwargs)
+        skip_checks = kwargs.pop("skip_input_checks", False)
+
+        if not skip_checks:
+            if "group" in kwargs:
+                cls._check_inputs(ref, hist, sim, group=kwargs["group"])
+
+            (ref, hist, sim), _ = cls._harmonize_units(ref, hist, sim)
+
+        out: xr.Dataset | xr.DataArray = cls._adjust(ref, hist, sim=sim, **kwargs)
+
+        if isinstance(out, xr.DataArray):
+            out = out.rename("scen").to_dataset()
+
+        scen = out.scen
+
+        params = ", ".join([f"{k}={v!r}" for k, v in kwargs.items()])
+        infostr = f"{cls.__name__}.adjust(ref, hist, sim, {params})"
+        scen.attrs["history"] = update_history(f"Bias-adjusted with {infostr}", sim)
+        scen.attrs["bias_adjustment"] = infostr
+
+        _is_multivariate = any(
+            [_crd.attrs.get("is_variables") for _crd in sim.coords.values()]
+        )
+        if _is_multivariate is False:
+            scen.attrs["units"] = ref.units
+
+        if OPTIONS[XSDBA_EXTRA_OUTPUT]:
+            return out
+        return scen
+
+
+class EmpiricalQuantileMapping(TrainAdjust):
+    """Empirical Quantile Mapping bias-adjustment.
+
+    Adjustment factors are computed between the quantiles of `ref` and `sim`.
+    Values of `sim` are matched to the corresponding quantiles of `hist` and corrected accordingly.
+
+    .. math::
+
+      F^{-1}_{ref} (F_{hist}(sim))
+
+    where :math:`F` is the cumulative distribution function (CDF) and `mod` stands for model data.
+
+    Attributes
+    ----------
+    Train step
+
+    nquantiles : int or 1d array of floats
+        The number of quantiles to use. Two endpoints at 1e-6 and 1 - 1e-6 will be added.
+        An array of quantiles [0, 1] can also be passed. Defaults to 20 quantiles.
+    kind : {'+', '*'}
+        The adjustment kind, either additive or multiplicative. Defaults to "+".
+    group : Union[str, Grouper]
+        The grouping information. See :py:class:`xsdba.base.Grouper` for details.
+        Default is "time", meaning an single adjustment group along dimension "time".
+    adapt_freq_thresh : str | None
+        Threshold for frequency adaptation. See :py:class:`xsdba.processing.adapt_freq` for details.
+        Default is None, meaning that frequency adaptation is not performed.
+
+    Adjust step:
+
+    interp : {'nearest', 'linear', 'cubic'}
+        The interpolation method to use when interpolating the adjustment factors. Defaults to "nearest".
+    extrapolation : {'constant', 'nan'}
+        The type of extrapolation to use. Defaults to "constant".
```

Comment:
it was already like this in `xclim`, I will check. 

There was  a problem though, duplicated entries in train/adjust cannot be repeated ... so the docstrings are incomplete to avoid this problem. It,s not like any docstring is complete elsewhere so I don't mind that much, but still a bit annoying

Reply:
seems all right:

![image](https://github.com/user-attachments/assets/12f43ec3-57f7-4fb2-9038-b5281271c9a6)

Reply:
The formatting makes `Train step` look like an Attribute.

One way around this is to document the attributes like so:
```
base_kws: dict = None
"""Arguments passed to blah blah"""

adj_kws: dict = None
"""Arguments passed to blah blah"""
```

Reply:
Surely there is a convention to document arguments of a method of an object (class) in the docstring?

Reply:
Python class documentation is tricky. Let me do some research.

Feel free to resolve this; It shouldn't block us making an initial release.